itests/net_test.go

```diff
@@ -0,0 +1,257 @@
+				assert.NotNil(t, stat.System)
+				assert.NotNil(t, stat.Transient)
+			case "system":
+				assert.NotNil(t, stat.System)
+			case "transient":
+				assert.NotNil(t, stat.Transient)
+			}
+		}
+	}
+
+	//stm: @NETWORK_COMMON_STAT_001
+	t.Run("all", withScope(t, "all"))
+	//stm: @NETWORK_COMMON_STAT_002
+	t.Run("system", withScope(t, "system"))
+	//stm: @NETWORK_COMMON_STAT_003
+	t.Run("transient", withScope(t, "transient"))
+	//stm: @NETWORK_COMMON_STAT_004
+	t.Run("peer", withScope(t, fmt.Sprintf("peer:%s", sId)))
+}
+
+func TestNetLimit(t *testing.T) {
+
+	firstNode, secondNode, _, _ := kit.EnsembleTwoOne(t)
+	ctx := context.Background()
+
+	sId, err := secondNode.ID(ctx)
+	require.NoError(t, err)
+
+	withScope := func(api interface{}, scope string) func(t *testing.T) {
+		return func(t *testing.T) {
+			_, err := firstNode.NetLimit(ctx, scope)
+			require.NoError(t, err)
+		}
+	}
+
+	//stm: @NETWORK_COMMON_LIMIT_001
+	t.Run("system", withScope(t, "system"))
+	//stm: @NETWORK_COMMON_LIMIT_002
+	t.Run("transient", withScope(t, "transient"))
+	//stm: @NETWORK_COMMON_LIMIT_003
+	t.Run("peer", withScope(t, fmt.Sprintf("peer:%s", sId)))
+}
+
+func TestNetBlockPeer(t *testing.T) {
+	ctx := context.Background()
+
+	firstNode, secondNode, _, ens := kit.EnsembleTwoOne(t)
+	ens.InterconnectAll()
+
+	//stm: @NETWORK_COMMON_ID_001
+	secondNodeID, err := secondNode.ID(ctx)
+	require.NoError(t, err)
+
+	//stm: @NETWORK_COMMON_BLOCK_ADD_001
+	err = firstNode.NetBlockAdd(ctx, api.NetBlockList{Peers: []peer.ID{secondNodeID}})
+	require.NoError(t, err)
+
+	//stm: @NETWORK_COMMON_BLOCK_LIST_001
+	list, err := firstNode.NetBlockList(ctx)
+	require.NoError(t, err)
+
+	if len(list.Peers) == 0 || list.Peers[0] != secondNodeID {
+		t.Errorf("blocked peer not in blocked peer list")
+	}
+
+	//stm: @NETWORK_COMMON_BLOCK_REMOVE_001
+	err = firstNode.NetBlockRemove(ctx, api.NetBlockList{Peers: []peer.ID{secondNodeID}})
+	require.NoError(t, err)
+
+	//stm: @NETWORK_COMMON_BLOCK_LIST_001
+	list, err = firstNode.NetBlockList(ctx)
+	require.NoError(t, err)
+
+	if len(list.Peers) > 0 {
+		t.Errorf("failed to remove blocked peer from blocked peer list")
+	}
+
+}
+
+func TestNetBlockIPAddr(t *testing.T) {
+	ctx := context.Background()
+
+	firstNode, secondNode, _, ens := kit.EnsembleTwoOne(t)
+	ens.InterconnectAll()
+
+	//stm: @NETWORK_COMMON_ADDRS_LISTEN_001
+	addrInfo, _ := secondNode.NetAddrsListen(ctx)
+
+	var secondNodeIPs []string
+
+	for _, addr := range addrInfo.Addrs {
+		ip, err := manet.ToIP(addr)
+		if err != nil {
+			continue
+		}
+		secondNodeIPs = append(secondNodeIPs, ip.String())
+	}
+
+	//stm: @NETWORK_COMMON_BLOCK_ADD_001
+	err := firstNode.NetBlockAdd(ctx, api.NetBlockList{IPAddrs: secondNodeIPs})
+	require.NoError(t, err)
+
+	//stm: @NETWORK_COMMON_BLOCK_LIST_001
+	list, err := firstNode.NetBlockList(ctx)
+	require.NoError(t, err)
+
+	if len(list.IPAddrs) == 0 || list.IPAddrs[0] != secondNodeIPs[0] {
+		t.Errorf("blocked ip not in blocked ip list")
+	}
```

Comment:
Can we try to NetConnect here, and see if it fails?

Reply:
Hey @magik6k , I asked about this part on slack but it got deleted, i'm not sure how this should work.
NetConnect won't fail there, nodes will still be able to connect, and that is true for peer,subnet and ip blocking.
Besides that, i tried to use full node and miner node, then block miner node and try to mine block and sync with full node, and even that works. 
So blocking a node seems to not do anything. Do you have any tips on how this should be tested?

Reply:
Added this check